app/assets/javascripts/components/AccountMenu/index.tsx

```diff
@@ -0,0 +1,163 @@
+  const [encryptionStatusString, setEncryptionStatusString] = useState<string | undefined>(undefined);
+  const [isEncryptionEnabled, setIsEncryptionEnabled] = useState(false);
+  const [server, setServer] = useState<string | undefined>(application.getHost());
+  const [isBackupEncrypted, setIsBackupEncrypted] = useState(isEncryptionEnabled);
+  const [protectionsDisabledUntil, setProtectionsDisabledUntil] = useState(getProtectionsDisabledUntil());
+  const [user, setUser] = useState(application.getUser());
+  const [hasProtections] = useState(application.hasProtectionSources());
+
+  const { notesAndTagsCount } = appState.accountMenu;
+
+  const closeAccountMenu = () => {
+    appState.accountMenu.closeAccountMenu();
+  };
+
+  // Add the required event observers
+  useEffect(() => {
+    const removeKeyStatusChangedObserver = application.addEventObserver(
+      async () => {
+        setUser(application.getUser());
+      },
+      ApplicationEvent.KeyStatusChanged
+    );
+
+    const removeProtectionSessionExpiryDateChangedObserver = application.addEventObserver(
+      async () => {
+        setProtectionsDisabledUntil(getProtectionsDisabledUntil());
+      },
+      ApplicationEvent.ProtectionSessionExpiryDateChanged
+    );
+
+    return () => {
+      removeKeyStatusChangedObserver();
+      removeProtectionSessionExpiryDateChangedObserver();
+    };
+  }, [application, getProtectionsDisabledUntil]);
+
+  return (
+    <div className="sn-component">
+      <div id="account-panel" className="sk-panel">
+        <div className="sk-panel-header">
+          <div className="sk-panel-header-title">Account</div>
+          <a className="sk-a info close-button" onClick={closeAccountMenu}>Close</a>
+        </div>
+        <div className="sk-panel-content">
+          <Authentication
+            application={application}
+            server={server}
+            setServer={setServer}
+            closeAccountMenu={closeAccountMenu}
+            notesAndTagsCount={notesAndTagsCount}
+            showLogin={showLogin}
+            setShowLogin={setShowLogin}
+            showRegister={showRegister}
+            setShowRegister={setShowRegister}
+            user={user}
+          />
+          {!showLogin && !showRegister && (
+            <div>
+              {user && (
+                <User
+                  email={user.email}
+                  server={server}
+                  appState={appState}
+                  application={application}
+                  closeAccountMenu={closeAccountMenu}
+                />
+              )}
+              <Encryption
+                isEncryptionEnabled={isEncryptionEnabled}
+                notesAndTagsCount={notesAndTagsCount}
+                encryptionStatusString={encryptionStatusString}
+              />
+              {hasProtections && (
+                <Protections
+                  application={application}
+                  protectionsDisabledUntil={protectionsDisabledUntil}
```

Comment:
If protectionsDisabledUntil is only used in the Protections component, we could move the related logic (such as the getProtectionsDisabledUntil method, the state variables and the related event observer) to that child component

Reply:
Done.